[FIX] fix create subscription request from partner

huguesdk opened the pull request:
## description

fix creation of subscription request from partner by adding missing fields: `firstname`, `lastname` and `lang`.

## odoo task

[task](https://gestion.coopiteasy.be/web#id=8408&view_type=form&model=project.task)

robinkeunen commented on easy_my_coop/models/subscription_request.py:
```diff
@@ -91,6 +91,17 @@ def _send_confirmation_mail(self):
+            if not vals.get("company_name"):
+                raise ValidationError(_("Company name must be defined for a company"))
+        else:
+            if not vals.get("firstname") or not vals.get("lastname"):
```
I find this more pythonic but it's a matter of taste

```py
if "firstname" not in vals or "lastname" not in vals:
```

huguesdk replied:
they could be in `vals` but as falsy values, which should also fail.

robinkeunen replied:
good point 👌

robinkeunen reviewed (approved):
taste comment, otherwise lgtm 👍

huguesdk committed:
[FIX] fix create subscription request from partner

fix creation of subscription request from partner by adding missing
fields: firstname, lastname and lang.

huguesdk commented:
i’ve just pushed a new version of the changes, which changes much less. it does not touch the name field, as even for companies the subscription request must have a first name and a last name (of the representative person). because subscription requests can have a first name, a last name and a company name at the same time, it’s more intuitive that the name field (which is used for search) contains the same value (first name and last name) for all subscription requests.

carmenbianca reviewed (approved):
LGTM.

I think that it'd be nicer if we universally had only one `name` field without distinguishing between first name and last name, but that's probably out-of-scope.

huguesdk commented:
i agree with the one `name` field. it’s simpler. i thought that it was like that before, and that it’s the addition of the `firstname` and `lastname` fields that caused the mayhem, but i’ve just checked and i was wrong. these fields are there since several years. the thing that has changed is that we tried to somehow deprecate the `name` field ([task](https://gestion.coopiteasy.be/web#id=5389&view_type=form&model=project.task), #177).

i know that almost all web forms separate first name and last name (and so people might be surprised if it’s not the case), but i don’t know whether there are legal requirements to store them separated.
